components/menu/menu-item-mixin.js

```diff
@@ -0,0 +1,146 @@
+export const MenuItemMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			first: { type: Boolean, reflect: true }, // set by d2l-menu
+			hasChildView: { type: Boolean, attribute: 'has-child-view' },
+			hidden: { type: Boolean, reflect: true },
+			last: { type: String, reflect: true }, // set by d2l-menu
+			role: { type: String, reflect: true },
+			tabindex: { type: Number, reflect: true, attribute: 'tab-index' },
+			text: String
+		};
+	}
+
+	constructor() {
+		super();
+		this.__keyCodes = {
+			ENTER: 13,
+			LEFT: 37,
+			RIGHT: 39,
+			SPACE: 32
+		};
+		this.__children = null;
+		this.role = 'menuitem';
+		this.tabindex = -1;
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+		this.__initializeItem();
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldValue, propName) => {
+			if (propName === 'hidden') {
+				this._onHidden();
+			}
+		});
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.addEventListener('blur', this.__onBlur);
```

Comment:
Consider wiring these up in `firstUpdated`?  I think the preference for event wire-up is:

1. in template using `@some-event` when wiring up to elements defined in template
2. in `firstUpdated` when wiring up to `this` - has advantage of deferring wire-up, but relies on element being garbage collected to remove listeners
3. in `connectedCallback` when wiring up to elements outside of local DOM scope, since we need to take control of removing the listeners

Reply:
Ah good to know! I think there are a few other places that wire these up in `connectedCallback` so those perhaps should be changed if we want to be consistent about this.